lib/compat/wordpress-6.4/theme-previews.php

```diff
@@ -0,0 +1,23 @@
+<?php
+/**
+ * Enable theme previews in the Site Editor for block themes.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Allow developers to customize the back to dashboard link in the Site Editor
+ *
+ * @param array $settings The editor settings.
+ * @return array The editor settings.
+ */
+
+function gutenberg_theme_preview_block_editor_settings_all( $settings ) {
+	$settings['__experimentalDashboardLink'] = 'themes.php';
```

Comment:
If we go this way, it will change [the site logo's link](https://github.com/WordPress/gutenberg/blob/1f2149b47a36a7b14805eb90a7612ce2ac18ca25/packages/edit-site/src/components/site-hub/index.js#L66), too. I thought this was not intuitive for the user, so I added a new `__experimentalThemePreviewBackLink`. What do you think? 👀 @scruffian @draganescu 
<img width="348" alt="Screen Shot 2023-09-11 at 10 46 31" src="https://github.com/WordPress/gutenberg/assets/5287479/defe4f49-71da-48ec-862e-a295be800b71">

Also, I think we need to change the text on hover, which currently says "Go to the Dashboard".

Reply:
These issues already exist in the current approach with `__experimentalDashboardLink`. I think what we should probably do is also add an option to customize the text as well, but I'm curious to hear what others think.

IMO themes.php is part of the dashboard but I'm not sure everyone would agree with that.

Reply:
The theme selection screen is part of the WP Admin - as is the dashboard screen. For previewing themes I don't think there should be two actions going to two different places. So I think that _when previewing_ both the W logo and the link text should go to the same place, as previewing is a modal experience and these are the escape hatches to back where the user was before.

It should be the case that the text can be also overriden, so that it's clear where the action will land the user.

Reply:
> For previewing themes I don't think there should be two actions going to two different places. So I think that when previewing both the W logo and the link text should go to the same place

Thanks for your comment. That makes sense to me 🙂